### PR TITLE
docs(semantic): document `SymbolTable`

### PR DESCRIPTION
**Pull request by DonIsaac**
Part of #5870

### CODE REVIEW

**Comment by DonIsaac** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5978?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5978** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5978?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5976** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5976?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_ast/src/ast/js.rs [hunk 1]
```diff
@@ -1736,7 +1745,41 @@ pub struct BindingRestElement<'a> {
...
+/// ```js
+/// // function.id is None
+/// // use function.r#type to check if a node is a function expression.
+/// const foo = function() { }
+/// ```
+///
+/// ```ts
+/// // Function overloads will not have a body
+/// function add(a: number, b: number): number; // <-- No body
+/// function add(a: string, b: string): string; // <-- No bod
```
Comment (daflyinbed) [inline crates/oxc_ast/src/ast/js.rs:1778]: no body ?

**Comment by overlookmotel** [comment]
Could you please move this into its own stack, rather than stacked on top of #5976 so can get it merged ASAP? #5976 may take a while to review.

Apart from the typo which @daflyinbed spotted, this looks good to me.

Just a note: I would have preferred if this PR had been split in 2 for (1) AST type docs and (2) `SymbolTable` docs. They wouldn't even need to be stacked on top of each other as they're independent changes. No need to split it up now, just saying for next time...

**Comment by DonIsaac** [comment]
Re-created in #5998

**Pull request closed without merging**